symphony-bdk-core/src/main/java/com/symphony/bdk/core/OboServiceFactory.java

```diff
@@ -0,0 +1,80 @@
+package com.symphony.bdk.core;
+
+import com.symphony.bdk.core.auth.AuthSession;
+import com.symphony.bdk.core.client.ApiClientFactory;
+import com.symphony.bdk.core.config.model.BdkConfig;
+import com.symphony.bdk.core.retry.RetryWithRecoveryBuilder;
+import com.symphony.bdk.core.service.message.OboMessageService;
+import com.symphony.bdk.core.service.stream.OboStreamService;
+import com.symphony.bdk.core.service.user.OboUserService;
+import com.symphony.bdk.gen.api.MessagesApi;
+import com.symphony.bdk.gen.api.RoomMembershipApi;
+import com.symphony.bdk.gen.api.ShareApi;
+import com.symphony.bdk.gen.api.StreamsApi;
+import com.symphony.bdk.gen.api.UserApi;
+import com.symphony.bdk.gen.api.UsersApi;
+import com.symphony.bdk.http.api.ApiClient;
+import com.symphony.bdk.http.api.ApiException;
+import com.symphony.bdk.template.api.TemplateEngine;
+
+import org.apiguardian.api.API;
+
+/**
+ * Factory responsible for creating BDK OBO-enabled service instances for {@link OboServicesFacade}.
+ * :
+ * <ul>
+ *   <li>{@link OboUserService}</li>
+ *   <li>{@link OboStreamService}</li>
+ *   <li>{@link OboMessageService}</li>
+ * </ul>
+ */
+@API(status = API.Status.INTERNAL)
+public class OboServiceFactory {
```

Comment:
Does this class need to be `public` (as it is only used by the `OboServicesFacade`)?  Could we make it package-protected?